src/ejabberd_app.erl

```diff
@@ -152,10 +152,20 @@ stop_modules() ->
 -spec start_services() -> ok.
 start_services() ->
     lists:foreach(
-        fun({Service, _}) -> mongoose_service:ensure_loaded(Service) end,
-        ejabberd_config:get_local_option_or_default(services, [])
+        fun({Service, Opts}) -> mongoose_service:ensure_loaded(Service, Opts) end,
+        services_with_maybe_metrics()
     ).
 
+-spec services_with_maybe_metrics() -> [term()].
+services_with_maybe_metrics() ->
```

Comment:
I don't think it's a good idea to add a service to a list of services if it was not there (according to the configuration). In my opinion it would be better to check if the `service_mongoose_system_metrics` is started and print appropriate log message if it is not. The printing can be done by calling an exported function from the service's module.